src/Stratis.Bitcoin.Features.PoA/PoAMiner.cs

```diff
@@ -127,34 +128,31 @@ private async Task CreateBlocksAsync()
 
                     this.logger.LogInformation("Waiting {0} seconds until block can be mined.", waitingTimeInSeconds );
 
-                    if (waitingTimeInSeconds  > 0)
+                    if (waitingTimeInSeconds > 0)
                     {
                         // Wait until we can mine.
-                        await Task.Delay(waitingTimeInSeconds  * 1000, this.cancellation.Token).ConfigureAwait(false);
+                        await Task.Delay(waitingTimeInSeconds * 1000, this.cancellation.Token).ConfigureAwait(false);
                     }
 
                     ChainedHeader tip = this.consensusManager.Tip;
 
-                    BlockTemplate blockTemplate = this.blockDefinition.Build(tip);
+                    Script walletScriptPubKey = this.GetScriptPubKeyFromWallet();
 
-                    // Premine.
-                    if (tip.Height + 1 == this.network.Consensus.PremineHeight)
+                    if (walletScriptPubKey == null)
                     {
-                        Script premineScriptPubKey = this.GetPremineScriptPubKey();
+                        this.logger.LogWarning("Miner wasn't able to get address from the wallet! You will not receive any rewards.");
+                        walletScriptPubKey = new Script();
+                    }
 
-                        if (premineScriptPubKey == null)
-                        {
-                            this.logger.LogWarning("Miner wasn't able to get address from the wallet to add premine reward!");
-                        }
-                        else
-                        {
-                            Transaction coinbase = blockTemplate.Block.Transactions[0];
+                    BlockTemplate blockTemplate = this.blockDefinition.Build(tip, walletScriptPubKey);
 
-                            coinbase.Outputs[0].ScriptPubKey = premineScriptPubKey;
-                            coinbase.Outputs[0].Value = this.network.Consensus.PremineReward;
+                    // Premine.
+                    if (tip.Height + 1 == this.network.Consensus.PremineHeight)
+                    {
+                        Transaction coinbase = blockTemplate.Block.Transactions[0];
+                        coinbase.Outputs[0].Value += this.network.Consensus.PremineReward;
```

Comment:
in block `blockDefinition.OnBuild()` we use this lines to get the premine
```
            var coinviewRule = this.ConsensusManager.ConsensusRules.GetRule<CoinViewRule>();
            this.coinbase.Outputs[0].Value = this.fees + coinviewRule.GetProofOfWorkReward(this.height);
```

Reply:
Yeah, but it's not PoW reward in our case. We wanted to refactor this at some point to be just reward.
But I can move premine there, no problem

Reply:
Fixed